### PR TITLE
[hue] Semantics added to channel types

### PR DESCRIPTION
**Pull request by lolodomo**
Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Review by Skinah — APPROVED** [review]
EDIT: @lolodomo  I had a second look after playing with tags the last few days....

 'Control' and 'Light' tags give the locations tabs a number of how many lights are turned ON in that location.. etc.
See this for example and note the bulb icon with a number of how many lights are on in that location...
http://www.pcmus.com/openhab/locations.png

Some of the channels are tagged with these and since I don't own any HUE I am not certain but some I would question like the effect channel. Can the effect be turned ON when the light is actually OFF? I also don't believe it will make any difference tagging the String channels with this. Perhaps err on the side of caution and only tag the ones that control the light actually being ON and OFF, otherwise the light icon may read the wrong number of lights turned on in the locations tab.

**Comment by jlaur** [comment]
@lolodomo - is this ready for being merged?

**Comment by lolodomo** [comment]
> @lolodomo - is this ready for being merged?

No because we have a veto of @kubawolanin regarding this approach.

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (kaikreuzer) [inline bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:30]: I'd prefer "Measurement" tags to be only used for Numbers with dimension, so that it is clear, what exactly is measured.

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (kaikreuzer) [inline bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:42]: Property tags are supposed to be used with "Control" and "Measurement" points, but not with "Status" (I know, this isn't clearly documented anywhere, sorry...).

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (kaikreuzer) [inline bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:53]: see above

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (kaikreuzer) [inline bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:158]: Actually, I think the semantic of this one should be "Alarm", not "Light". One uses it in case one want to signal an alarm and not in order to have light in the room.

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (kaikreuzer) [inline bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:177]: Sending "ON" to a "Control-Light" semantic item, I would expect to turn on some light (to make the room lit) and not to start some fancy effect.

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (kaikreuzer) [inline bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:201]: From a machine-readability point, I also think those tags don't have much meaning, since a machine can't make any sense from a String item state in terms of what this means for a light.

**Review by kaikreuzer — CHANGES_REQUESTED** [review]
> No because we have a kind of veto from @kubawolanin regarding this approach.

You probably wanted to mention me instead here.
As commented on the other PRs, I would prefer to reduce this to the "obvious" fits.
As @Skinah mentioned above, the tags in this PR might have some unexpected effects to users.
I left some comments below on channels, which I consider problematic.

**Comment by lolodomo** [comment]
I am going to consider review comments.

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:30]: But what else to use in this case ?

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:42]: Tags removed

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:53]: Tags removed

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:158]: Updated to use Alarm

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:177]: Tags removed

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml]: Tags removed

**Comment by lolodomo** [comment]
@kaikreuzer : I updated the PR, remains only one open point to discuss

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (kaikreuzer) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:30]: Following our agreement to only tag what is obvious, I'd vote to not tag it at all.

Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml]: Ok, this is now removed.

**Comment by lolodomo** [comment]
@kaikreuzer : last change done

File: bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml
Comment (jlaur) [inline bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:146]: Is this combination valid?

Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml:146]: You're right, I was just checking the semantics, they are both "Point".
I am going to suppress "Control".

Comment (lolodomo) [reply bundles/org.openhab.binding.hue/src/main/resources/OH-INF/thing/channels.xml]: Fixed

**Comment by lolodomo** [comment]
@kaikreuzer : I am counting 27 bindings defining tags on some of their channels. Maybe you could check that they all define tags properly.

1. airquality
2. danfossairunit
3. deconz
4. dmx
5. feican
6. homeconnect
7. ipobserver
8. lifx
9. max
10. mihome
11. milight
12. millheat
13. mqtt.espmilighthub
14. nikohomecontrol
15. openweathermap
16. openwebnet
17. qbus
18. rotel
19. sagercaster
20. sensibo
21. shelly
22. sonyprojector
23. souliss
24. touchwand
25. vigicrues
26. wled
27. yeelight

**Comment by jlaur** [comment]
@lolodomo - would you like code owner (@cweitkamp), or @kaikreuzer to review before merging? It looks good to me now after removing the tags which were not straight-forward.

**Review by jlaur — APPROVED** [review]
LGTM.

**Comment by lolodomo** [comment]
> @lolodomo - would you like code owner (@cweitkamp), or @kaikreuzer to review before merging? It looks good to me now after removing the tags which were not straight-forward.

No, I don't think this is necessary, there is now almost nothing changed and Kai already made his comments.

**Pull request merged**
